data/src/main/java/me/androidbox/data/worker_manager/TaskCreateSynchronizerWorker.kt

```diff
@@ -0,0 +1,78 @@
+package me.androidbox.data.worker_manager
+
+import android.content.Context
+import androidx.hilt.work.HiltWorker
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.async
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.supervisorScope
+import me.androidbox.data.local.dao.TaskDao
+import me.androidbox.data.local.entity.TaskEntity
+import me.androidbox.data.mapper.toTaskDto
+import me.androidbox.data.remote.network.task.TaskService
+import me.androidbox.data.remote.util.CheckResult.checkResult
+import me.androidbox.data.worker_manager.constant.Constant.RETRY_COUNT
+import me.androidbox.domain.constant.SyncAgendaType
+
+@HiltWorker
+class TaskCreateSynchronizerWorker @AssistedInject constructor(
+    @Assisted private val context: Context,
+    @Assisted private val workerParameters: WorkerParameters,
+    private val taskService: TaskService,
+    private val taskDao: TaskDao
+) : CoroutineWorker(context, workerParameters) {
+
+    override suspend fun doWork(): Result {
+        val createdTasks: Deferred<List<TaskEntity>>
+
+        /** Get all the created tasks */
+        supervisorScope {
+            createdTasks = async {
+                taskDao.getAllTasksBySyncType(SyncAgendaType.CREATE).map {
+                    it.task
+                }
+            }
+        }
+
+        val result = if (createdTasks.await().isNotEmpty()) {
+            if(runAttemptCount > RETRY_COUNT) {
+                Result.failure()
+            }
+            else {
+                Result.retry()
+            }
```

Comment:
If we have created tasks that need to be sync'ed. If there is a failure. Is it ok to do it in this block?

Reply:
Not really getting what this is for... Don't you want to sync these tasks here? Why return a failure if there are tasks to sync?

Reply:
@philipplackner Maybe a miss understanding. I am wondering is this the best place for this:
```
            if(runAttemptCount > RETRY_COUNT) {
                Result.failure()
            }
            else {
                Result.retry()
            }
```

Reply:
Personally I'd add it at the start of the worker. So you can avoid calling the dao uselessly

Reply:
@MKiperszmid I have updated, Can you confirm this is correct with what I have done?